### PR TITLE
chore: update codecov configuration

### PR DESCRIPTION
**Pull request by armano2**
Update codecov to not wait for all ci to report its status, this should fix issue with 

> Expected — Waiting for status to be reported

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -142,7 +142,6 @@ jobs:
       - name: Publish code coverage report
         uses: codecov/codecov-action@v1
         with:
-          yml: ./codecov.yml
```
Comment (armano2) [inline .github/workflows/ci.yml:145]: > Warning: Unexpected input(s) 'yml', valid inputs are ['token', 'files', 'directory', 'flags', 'aws_curl_args', 'codecov_curl_args', 'commit_parent', 'env_vars', 'fail_ci_if_error', 'file', 'functionalities', 'gcov_args', 'gcov_executable', 'gcov_path_exclude', 'gcov_path_include', 'gcov_prefix', 'gcov_root_dir', 'move_coverage_to_trash', 'name', 'override_branch', 'override_build', 'override_commit', 'override_pr', 'override_tag', 'path_to_write_report', 'root_dir', 'verbose', 'working-directory', 'xcode_derived_data', 'xcode_package']

**Review by bradzacher — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -60,6 +60,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
```
Comment (armano2) [inline .github/workflows/ci.yml:64]: this is needed for codecov to be able to detect correct commit SHA

> ==> GitHub Actions detected.
> ->  Issue detecting commit SHA. Please run actions/checkout with fetch-depth > 1 or set to 0

Comment (bradzacher) [reply .github/workflows/ci.yml:64]: could you set it to `0`?

Comment (armano2) [reply .github/workflows/ci.yml:64]: we could but its better not to, fetching only 1 commit is huge boost to time of git fetch/clone

this property is used to tell git how many commit we want to fetch

**Review by bradzacher — APPROVED** [review]

**Pull request merged**